alerts/resource_alerts.libsonnet

```diff
@@ -57,7 +57,7 @@
           {
             alert: 'KubeCPUQuotaOvercommit',
             expr: |||
-              sum(kube_resourcequota{%(prefixedNamespaceSelector)s%(kubeStateMetricsSelector)s, type="hard", resource="cpu"})
+              sum(kube_resourcequota{%(prefixedNamespaceSelector)s%(kubeStateMetricsSelector)s, type="hard", resource=~"(cpu|requests.cpu)"})
```

Comment:
According to https://kubernetes.io/docs/concepts/policy/resource-quotas/#compute-resource-quota `cpu` and `requests.cpu` should be the same. However, I am not sure if they can coexist, because if they can, this alert would be incorrect due to summing duplicated data.

Reply:
That's indeed a must to test corner case. Will get back to you soon :) Thanks for your review @paulfantom .

Reply:
I did some analysis around this.

Resource names/limits are just key value pairs. So a ResourceQuota can have both `cpu` and `requests.cpu` specified under `.spec.hard` attribute.

When a Pod created under a namespace, resourcequota [admission controller checks](https://github.com/kubernetes/kubernetes/blob/ea0764452222146c47ec826977f49d7001b0ea8c/staging/src/k8s.io/apiserver/pkg/admission/plugin/resourcequota/controller.go#558) whether the pod has enough resource from the specified quota. For the comparison Pod seem to be providing both `cpu` and `requests.cpu` with same value from it's `.spec.containers[].resources.requests.cpu`.

For example, when you have both `cpu` and `requests.cpu` defined in the quota with different values, the comparison looks like below,

```
Quota{cpu="7m", requests.cpu="9m"} < Request{cpu="8m", requests.cpu="8m"}
```
The above comparison will fail.

So my conclusion is "we should consider the min value into account when quota has both `cpu` and `requests.cpu`. WDYT?

Reply:
Thanks for an explanation from kubernetes side. I just still wonder how that would transfer into metics with kube-state-metrics. Especially is the following is possible:
```
kube_resourcequota{pod="A",type="hard", resource="cpu"}
kube_resourcequota{pod="A",type="hard", resource="requests.cpu"}
```
And what would be the possible values and would they differ? 

Also how `Quota{cpu="7m", requests.cpu="9m"}` is possible? Docs say that `cpu` and `requests.cpu` are the same :thinking:

Reply:
`Quota{cpu="7m", requests.cpu="9m"}` is very much possible, look at https://gist.github.com/arajkumar/d337965c92bc674d90aea81d4c7ea6e9#file-quota-yaml-L38. It is considered as is from the ResourceQuota definition.

Also I've added tests for this., hope that explains.!